.github/workflows/ci.yaml

```diff
@@ -0,0 +1,89 @@
+name: Build snap and run tests
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        required: true
+        type: string
+    outputs:
+      snap-file:
+        description: "The snap output of build process."
+        value: ${{ jobs.build.outputs.snap-file }}
+
+jobs:
+  check:
+    name: Check build condition for Snap
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version:
+          - '3.9'
+          - '3.10'
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{matrix.python-version}}
+          architecture: x64
+      - id: code-checks
+        name: Make sure the code follows conventions and tests correctly
+        run: |
+          make checks
+  build:
+    name: Build Snap
+    runs-on: ubuntu-latest
+    needs:
+      - check
+    outputs:
+      snap-file: ${{ steps.build-snap.outputs.snap }}
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+      - id: build-snap
+        name: Build snap
+        uses: snapcore/action-build@v1
+        with:
+          snapcraft-channel: 7.x/candidate
```

Comment:
FJMI, is there some fixes for you in 7.2.10?

Reply:
Not expecting any. Can change this in future.